proc/groupby.go

```diff
@@ -161,26 +213,99 @@ func NewGroupByAggregator(c *Context, params GroupByParams) *GroupByAggregator {
 		}
 		rs := expr.NewCompareFn(true, expr.CompileFieldAccess(params.keys[0].target))
 		if params.inputSortDir < 0 {
-			recordCompare = func(a, b *zng.Record) int { return rs(b, a) }
+			keyCompare = func(a, b *zng.Record) int { return rs(b, a) }
 		} else {
-			recordCompare = rs
+			keyCompare = rs
 		}
 	}
+	var resolvers []expr.FieldExprResolver
+	for _, k := range params.keys {
+		resolvers = append(resolvers, expr.CompileFieldAccess(k.target))
+	}
+	rs := expr.NewCompareFn(true, resolvers...)
+	if params.inputSortDir < 0 {
+		keysCompare = func(a, b *zng.Record) int { return rs(b, a) }
+	} else {
+		keysCompare = rs
+	}
+	sm, err := newSpillManager()
+	if err != nil {
+		c.Logger.Warn("groupby: could not create spill manager", zap.Error(err))
+	}
```

Comment:
This call creates a tempDir. I don't we should be creating a temp dir until a groupby needs to spill to disk.

Reply:
good point, will fix